### DISCUSSION TITLE
[Support]: Frigate 14 doesn't appear to report correct storage use?

### DISCUSSION THREAD

**Discussion by LaurenceGough** [blakeblackshear/frigate#13358] [General Support] [answered]
### Describe the problem you are having

Hi all,
 
My 1TB CCTV HDD is full - it is a dedicated drive to Frigate installed via SATA in my server. It is only used by the Frigate docker container. Frigate doesn't appear to be reporting the correct amount in use in the system metrics unless I'm doing something wrong?

Under "Camera Storage" it is listing my server's primary SSD  "/dev/root" as the total capacity instead. I believe this should be using the directory where CCTV recording storage is set to "/mnt/cctv" in my case? However it does display the total drive space correctly in "Recordings" but it doesn't show the correct space in use, instead showing the full capacity of the primary SSD as the in use capacity of the mounted 1TB HDD /mnt/cctv" which feels odd.

```
df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       234G   33G  190G  15% /
devtmpfs        3.8G     0  3.8G   0% /dev
tmpfs           3.8G     0  3.8G   0% /dev/shm
tmpfs           1.6G  1.7M  1.5G   1% /run
tmpfs           4.0M     0  4.0M   0% /sys/fs/cgroup
efivarfs        192K   72K  116K  39% /sys/firmware/efi/efivars
tmpfs           3.8G     0  3.8G   0% /tmp
/dev/sdb1       916G  904G  2.9G 100% /mnt/cctv
clr_debug_fuse  234G   33G  190G  15% /usr/src/debug
clr_debug_fuse  234G   33G  190G  15% /usr/lib/debug
tmpfs           769M  4.0K  769M   1% /run/user/1000
```

"/dev/sdb1       916G  904G  2.9G 100% /mnt/cctv" is the CCTV HDD and it is full as it says in the Frigate logs:

```info
2024-08-26 11:40:41
frigate.storage
Cleaned up 2784.100000000004 MB of recordings
info
2024-08-26 12:45:41
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 12:45:41
frigate.storage
Cleaned up 2785.1000000000017 MB of recordings
info
2024-08-26 13:45:41
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 13:45:42
frigate.storage
Cleaned up 2784.199999999998 MB of recordings
info
2024-08-26 14:45:42
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 14:45:43
frigate.storage
Cleaned up 2786.600000000001 MB of recordings
```

![Screenshot 2024-08-26 160129](https://github.com/user-attachments/assets/b50488f6-fc07-4115-a225-dd8eedc193f6)

Here is my Docker config file:

```
     - /etc/localtime:/etc/localtime:ro
      - /var/lib/docker/volumes/frigatevolume/config:/config
      - /var/lib/docker/volumes/frigatevolume/media/:/media/frigate
      - /mnt/cctv/:/media/frigate/recordings
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
```

Manually browsing the CCTV drive I can see a fair few folders from June and July this year which is beyond my retention periods set in the config. However trying to view these events in the history viewer for the cameras shows no footage on the timeline or event viewer "No events found for this time period." (no filters on). I believe this is expected due to the upgrade to Frigate 14.

Viewing these folders manually shows limited footage - it appears to be just zone trigger events - which does have a longer retention period but not this long.

I am not sure how event's files are dealt with in the upgrade to Frigate 14 but could people be having old events taking a lot of storage without them knowing?

Has the storage use per an hour per a camera been removed in Frigate 14, or should this what my "Camera Storage" section should be showing?

I have manually deleted some of the older folders for now to free up some space.

Please let me know if there is any more information or tests you'd like me to run.

Thank you.

### Version

0.14.0-da913d8

### Frigate config file

```yaml
Priv but can share if required.
```

### Relevant Frigate log output

```shell
info
2024-08-26 11:40:41
frigate.storage
Cleaned up 2784.100000000004 MB of recordings
info
2024-08-26 12:45:41
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 12:45:41
frigate.storage
Cleaned up 2785.1000000000017 MB of recordings
info
2024-08-26 13:45:41
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 13:45:42
frigate.storage
Cleaned up 2784.199999999998 MB of recordings
info
2024-08-26 14:45:42
frigate.storage
Less than 1 hour of recording space left, running storage maintenance...
info
2024-08-26 14:45:43
frigate.storage
Cleaned up 2786.600000000001 MB of recordings
```

### Relevant go2rtc log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Other Linux

### Install method

Docker Compose

### Object Detector

OpenVino

### Network connection

Wired

### Camera make and model

Hikvisions

### Screenshots of the Frigate UI's System metrics pages

![Screenshot 2024-08-26 160129](https://github.com/user-attachments/assets/c2a8f93c-7fb2-41e3-b0b1-83da5d72c49f)

**Comment by NickM-27** [comment]
The recordings part here just shows what is in the database, if you have other files in the same mount or files which are frigate recordings but were not properly deleted then those will not be counted

**Reply by LaurenceGough** [reply]
Hi Nick,

I've just had a look, I also had another person look to make sure I'm not going crazy and it appears that the retention period is being ignored both on the record retention and the event retention.

I have absolutely loads of recordings for July for example.

Here is an example of one of my camera's record configuration:

```
    record:
      enabled: true
      retain:
        days: 10
        mode: all
      events:
        pre_capture: 12
        post_capture: 15
        retain:
          default: 14
          mode: active_objects
          objects:
            person: 24
```

I believe this configuration is correct and no events or recordings should be stored after a month? Forget manually stared events, we only had a few and I have found them (from 2023).

How come these were not in the database, would it be from Upgrading to Frigate 14 which I did on 14/07/2024. I guess this means people will have to manually clean up on upgrading to Frigate 14?

**Reply by NickM-27** [reply]
it is not being ignored. Based on what was posted in this original issue, we know that these recordings are most likely not in the DB hence frigate does not know they exist. this can happen in a few different cases, they should just be deleted manually

**Reply by LaurenceGough** [reply]
Ok, thanks, I will just manually delete them.

Forgetting this, what about the other issues with the way the storage is being displayed in this stats screen which was the main thing I wished to report?

**Reply by NickM-27** [reply, marked as answer]
you are misunderstanding how this is working. Recordings shows `total space of recordings based on DB / total space of drive for recordings` and each camera item shows `space of recordings based on DB specific to that camera / total space of recordings based on DB`

**Reply by LaurenceGough** [reply]
Thanks Nick.
